add PlatformIdentifier component to contributing docs (#4450)
getsentry/sentry-docs: src/docs/contributing/pages/components.mdx

Note: This currently causes issues with tableOfContents generation, so its recommended to disable the TOC when using it.

## PlatformIdentifier

> Render terms in the correct case, for each platform, within the body text of a page (not in code samples):

Author: ```suggestion
Render terms in the correct case within the body text of a page (not in code samples) based on the platform case_style setting:
```

> Enforces the correct casing or case_style setting of terms for each platform in the main body text (not in code samples):

Author: ```suggestion
Render terms in the correct case, for each platform, within the body text of a page (not in code samples):
```

```markdown {tabTitle:Example}
<PlatformIdentifier name="before-send" />
```

> When you use <PlatformIdentifier name="before-send" />, it will be converted to:

Author: ```suggestion
For example, if you use <PlatformIdentifier name="before-send" />, it will render as:
```

> For example, if you use <PlatformIdentifier name="before-send" />, it will render as:

Author: ```suggestion
For example, if you use `<PlatformIdentifier name="before-send" />`, it will render as:
```

- `beforeSend` if case_style=camelCase
- `before_send` if case_style=snake_case
- `BeforeSend` if case_style=PascalCase

> This component only works in platform pages.

Author: ```suggestion
This component enforces the case_style setting of the platform and only works properly in platform pages.
```

> This component enforces the case_style setting of the platform and only works properly in platform pages.

Author: ```suggestion
This component only works properly in platform pages.
```
